src/crawlee/browsers/_playwright_browser_controller.py

```diff
@@ -152,8 +143,45 @@ async def _create_browser_context(self, proxy_info: ProxyInfo | None = None) ->
             else None
         )
 
-        return await self._browser.new_context(
+        if self._use_fingerprints:
```

Comment:
(This can be simpler if we decide to rely purely on browserforge and remove our own header generation. )